lib/tree-view.coffee

```diff
@@ -546,23 +546,32 @@ class TreeView extends View
           buttons: ['OK']
         return
 
+    if selectedPaths.length is 1 and fs.isFileSync(selectedPaths[0])
+      if fs.getSizeSync(selectedPaths[0]) is 0
+        @moveToTrash(selectedPaths)
+        return
+
+    moveToTrash = @moveToTrash
```

Comment:
Not necessary. Make the arrow in the `atom.confirm` button callback below a "fat arrow" to capture the `this` scope, then delete this line.